pkg/controller/cronjob/utils.go

```diff
@@ -63,28 +63,19 @@ func deleteFromActiveList(sj *batchv2alpha1.CronJob, uid types.UID) {
 
 // getParentUIDFromJob extracts UID of job's parent and whether it was found
 func getParentUIDFromJob(j batchv1.Job) (types.UID, bool) {
-	creatorRefJson, found := j.ObjectMeta.Annotations[v1.CreatedByAnnotation]
-	if !found {
-		glog.V(4).Infof("Job with no created-by annotation, name %s namespace %s", j.Name, j.Namespace)
-		return types.UID(""), false
-	}
-	var sr v1.SerializedReference
-	err := json.Unmarshal([]byte(creatorRefJson), &sr)
-	if err != nil {
-		glog.V(4).Infof("Job with unparsable created-by annotation, name %s namespace %s: %v", j.Name, j.Namespace, err)
+	controllerRef := controller.GetControllerOf(&j)
+
+	if controllerRef == nil {
+		glog.V(4).Infof("Failed to obtain controller reference, name %s namespace %s", j.Name, j.Namespace)
```

Comment:
As a side note, CronJob controller lists all jobs, and then find the jobs that are controlled by cronjobs. https://github.com/kubernetes/kubernetes/blob/v1.7.0-beta.2/pkg/controller/cronjob/cronjob_controller.go#L128

Therefore, it's normal to find a job without controller reference here, and thus no need to mention "failed" or "error" in the log.